tests/step-103.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+# Test:
+#   Fail on not available shared hooks.
+
+if ! sh /var/lib/githooks/install.sh; then
+    echo "! Failed to execute the install script"
+    exit 1
+fi
+
+mkdir -p /tmp/shared/hooks-102.git/pre-commit &&
+    echo 'exit 0' >/tmp/shared/hooks-102.git/pre-commit/succeed &&
+    cd /tmp/shared/hooks-102.git &&
+    git init &&
+    git add . &&
+    git commit -m 'Initial commit' ||
+    exit 1
+
+# Install shared hook url into a repo.
+mkdir -p /tmp/test102 && cd /tmp/test102 || exit 1
+git init || exit 1
+mkdir -p .githooks && echo '/tmp/shared/hooks-102.git' >.githooks/.shared || exit 1
+git add .githooks/.shared
+git hooks shared update
+
+RESULT=$(ls -A ~/.githooks/shared 2>/dev/null)
+if [ "$RESULT" = "" ]; then
+    echo "! Expected shared hooks to be installed."
+    exit 1
+fi
+git commit -m "Test" || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
+
+# Clone a new one
+echo "Cloning"
+cd /tmp || exit 1
+git clone /tmp/test102 test102-clone && cd test102-clone || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
+
+echo "Commiting"
+# Make a commit
+echo A >A || exit 1
+git add A || exit 1
+OUTPUT=$(git commit -a -m "Test" 2>&1)
+
+# shellcheck disable=SC2181
+if [ $? -eq 0 ] || ! echo "$OUTPUT" | grep -q "Failed to execute shared hook"; then
```

Comment:
The commit should be successful, and that's how it should be.
This may produce some extra message about non-existing shared hooks which could be suppressed, but I don't think the commit should fail because of it.

Reply:
I strongly think: you placed a `.shared` file into your repo. Meaning you want hooks to be applied. When you clone, the shared hooks are not there. You need manual `git hooks shared update`. Which can get forgotten. 
I made it such that the commit fails, because shared hooks are not present...

Reply:
The problem with these optional behavior is, that user forget things, and I think we should not give the user to much space to really mess up things... git is hard enough :-)...